packages/wrangler/src/pages/deploy.tsx

```diff
@@ -337,9 +345,65 @@ export const Handler = async (args: PagesDeployArgs) => {
 		project_name: projectName,
 	});
 
-	logger.log(
-		`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
-	);
+	let latestDeploymentStage: DeploymentStage | undefined;
+	let attempts = 0;
+
+	logger.log("🌎 Checking deployment status...");
+
+	while (
+		attempts < MAX_DEPLOYMENT_STATUS_ATTEMPTS &&
+		latestDeploymentStage?.name !== "deploy" &&
+		latestDeploymentStage?.status !== "success" &&
+		latestDeploymentStage?.status !== "failure"
+	) {
+		try {
+			/*
+			 * Exponential backoff
+			 * On every retry, exponentially increase the wait time: 1 second, then
+			 * 2s, then 4s, then 8s, etc.
+			 */
+			await new Promise((resolvePromise) =>
+				setTimeout(resolvePromise, Math.pow(2, attempts++) * 1000)
+			);
+
+			logger.debug(
+				`attempt #${attempts}: Attempting to fetch status for deployment with id "${deploymentResponse.id}" ...`
+			);
+
+			const deployment = await fetchResult<Deployment>(
+				`/accounts/${accountId}/pages/projects/${projectName}/deployments/${deploymentResponse.id}`
+			);
+			latestDeploymentStage = deployment.latest_stage;
+		} catch (err) {
+			// don't retry if API call retruned an error
+			logger.debug(
+				`Attempt to get deployment status for deployment with id "${deploymentResponse.id}" failed: ${err}`
+			);
+		}
+	}
+
+	if (latestDeploymentStage?.status === "success") {
+		logger.log(
+			`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
+		);
+	} else if (latestDeploymentStage?.status === "failure") {
+		// get persistent logs so we can show users the failure message
+		const logs = await fetchResult<UnifiedDeploymentLogMessages>(
+			`/accounts/${accountId}/pages/projects/${projectName}/deployments/${deploymentResponse.id}/history/logs?size=10000000`
+		);
+		// last log entry will be the most relevant for Direct Uploads
+		const failureMessage = logs.data[logs.total - 1].line;
+
+		logger.error(failureMessage.replace("Error:", "").trim());
+		logger.log("❌ Deployment failed!");
+	} else {
+		logger.log(
+			`✨ Deployment complete! However, we couldn't ascertain the final status of your deployment.\n\n` +
+				`⚡️ Visit your deployment at ${deploymentResponse.url}\n` +
+				`⚡️ Check the deployment logs on the Cloudflare dashboard: https://dash.cloudflare.com/${accountId}/pages/view/${projectName}/${deploymentResponse.id}`
```

Comment:
There are gonna be no logs here for DU users, they could see a green tick and red triangle but that's it. I probably wouldn't bother linking to dash

Reply:
true, but given that we couldn't tell whether the deployment was successful or failed from within wrangler, I think there might be some value in linking to Dash. I changed the wording to `Check the deployment details on ...`